Scripts/Haptics/HapticPulseFromAudioClip.cs

```diff
@@ -0,0 +1,121 @@
+﻿namespace VRTK.Core.Haptics
+{
+    using UnityEngine;
+    using System.Collections;
+
+    /// <summary>
+    /// Creates a haptic pattern based on the waveform of an <see cref="AudioClip"/> and utilizes a <see cref="HapticPulse"/> to create the effect.
+    /// </summary>
+    public class HapticPulseFromAudioClip : HapticProcess
+    {
+        /// <summary>
+        /// The pulse process to utilize.
+        /// </summary>
+        [Tooltip("The pulse process to utilize.")]
+        public HapticPulse hapticPulse;
+        /// <summary>
+        /// The waveform to represent the haptic pattern.
+        /// </summary>
+        [Tooltip("The waveform to represent the haptic pattern."), SerializeField]
+        protected AudioClip audioClip;
+        /// <summary>
+        /// Multiplies the current audio peak to affect the wave intensity.
+        /// </summary>
+        [Tooltip("Multiplies the current audio peak to affect the wave intensity."), SerializeField]
+        protected float intensityMultiplier = 1f;
+
+        /// <summary>
+        /// The waveform to represent the haptic pattern.
+        /// </summary>
+        public AudioClip AudioClip
+        {
+            get
+            {
+                return audioClip;
+            }
+            set
+            {
+                audioClip = value;
+            }
+        }
+
+        /// <summary>
+        /// Multiplies the current audio peak to affect the wave intensity.
+        /// </summary>
+        public float IntensityMultiplier
+        {
+            get { return intensityMultiplier; }
+            set { intensityMultiplier = value; }
+        }
+
+        protected Coroutine hapticRoutine;
+        /// <summary>
+        /// The original intensity of the provided <see cref="HapticPulse"/> to reset back to after the process is complete.
+        /// </summary>
+        protected float cachedIntensity;
+        /// <summary>
+        /// The size of the audio buffer.
+        /// </summary>
+        protected int bufferSize = 8192;
+
+        /// <inheritdoc />
+        public override bool IsActive()
+        {
+            return (isActiveAndEnabled && hapticPulse != null && audioClip != null && hapticPulse.isActiveAndEnabled);
```

Comment:
After the move of `isActiveAndEnabled` to the base this should read `return (base.IsActive() && hapticPulse != null && audioClip != null && hapticPulse.IsActive());`.

Note that last `hapticPulse.IsActive()` which should probably be a thing even if we're not moving anything to the base.